contracts/test/StakeLocker.t.sol

```diff
@@ -2,663 +2,254 @@
-
-        // Still within Staker unstake window
-        hevm.warp(start + globals.stakerCooldownPeriod() + 1);
-        assertTrue(sam.try_unstake(address(stakeLocker1), amt), "Should be able to unstake funds again during cooldown window");
+        assertEq(address(locker.stakeAsset()),     stakeAsset,     "Incorrect address of stake asset");
+        assertEq(address(locker.liquidityAsset()), liquidityAsset, "Incorrect address of liquidity asset");
+        assertEq(locker.pool(),                    pool,           "Incorrect address of pool");
+        assertEq(locker.lockupPeriod(),            180 days,       "Incorrect lockup period");
+    }
 
-        // Second after Staker unstake window ends
-        hevm.warp(start + globals.stakerCooldownPeriod() + globals.stakerUnstakeWindow() + 1);
-        assertTrue(!sam.try_unstake(address(stakeLocker1), amt), "Should fail to unstake funds because now past unstake window");
+}
 
-        uint256 newStart = block.timestamp;
+contract StakeLockerStakerTest is DSTest {
 
-        // Intend to unstake
-        assertTrue(sam.try_intendToUnstake(address(stakeLocker1)), "Failed to intend to unstake");
+    Hevm hevm;
 
-        // After cooldown ends but after unstake window
-        hevm.warp(newStart + globals.stakerCooldownPeriod() + globals.stakerUnstakeWindow() + 1);
-        assertTrue(!sam.try_unstake(address(stakeLocker1), amt), "Should fail to unstake after unstake window has passed");
+    StakeLockerPoolDelegate internal delegate;
+    MapleGlobalsMock        internal globals;
+    PoolFactoryMock         internal factory;
+    PoolMock                internal pool;
+    MockToken               internal stakeToken;
+    MockToken               internal liquidityToken;
+    StakeLocker             internal locker;
+    StakeLockerStaker       internal staker;
 
-        // Last second of Staker unstake window
-        hevm.warp(newStart + globals.stakerCooldownPeriod() + globals.stakerUnstakeWindow());
-        assertTrue(sam.try_unstake(address(stakeLocker1), amt), "Should be able to unstake during unstake window");
+    constructor() public {
+        hevm = Hevm(address(bytes20(uint160(uint256(keccak256("hevm cheat code"))))));
     }
 
-    function test_stake_transfer_paused() public {
-        makePublicAndStake(25 * WAD);
-        pat.setStakeLockerLockupPeriod(address(stakeLocker1), 0);
-
-        // Pause protocol and attempt to transfer FDTs
-        assertTrue( emergencyAdmin.try_setProtocolPause(address(globals), true));
-        assertTrue(!sam.try_transfer(address(stakeLocker1), address(leo), 1 * WAD));
-
-        // Unpause protocol and transfer FDTs
-        assertTrue(emergencyAdmin.try_setProtocolPause(address(globals), false));
-        assertTrue(sam.try_transfer(address(stakeLocker1), address(leo), 1 * WAD));
+    function constrictToRange(uint256 value, uint256 min, uint256 max) internal pure returns (uint256) {
+        return value % (max - min) + min;        
     }
```

Comment:
From test util: if you can figure out a better way to deal with (0, 0) then I'm totally open to it
```js
    function constrictToRange(uint256 val, uint256 min, uint256 max) public pure returns (uint256) {
        return constrictToRange(val, min, max, false);
    }

    function constrictToRange(uint256 val, uint256 min, uint256 max, bool nonZero) public pure returns (uint256) {
        if      (val == 0 && !nonZero) return 0;
        else if (max == min)           return max;
        else                           return val % (max - min) + min;
    }
```

Reply:
I don't understand. Just don't call `constrictToRange` with a min and max that isn't valid in your use case.